src/proc/alu_private.c

```diff
@@ -36,44 +36,44 @@ static U32 div2(U16 lhs, U16 rhs) {
   return ((remainder << 16) & 0xFFFF0000) | (result & 0x0000FFFF);
 }
 
-static void compute(Private_ALU *self, BinaryOperator op, Register *lhs, Register *rhs, OverflowConsumer consumer) {
+static void compute(Private_ALU *self, BinaryOperator op, Register2 *lhs, Register2 *rhs, OverflowConsumer consumer) {
```

Comment:
```suggestion
static void compute(Private_ALU *self, BinaryOperator op, U16 *lhs, U16 *rhs, OverflowConsumer consumer) {
```

Besides just renaming, I would recommend changing RHS to be passed-by-value